spec/mongoid/query_cache_spec.rb

```diff
@@ -452,6 +452,19 @@
       Band.collection.indexes.create_one(name: 1)
     end
   end
+
+  context 'when the initial query does not exhaust the results' do
+    before do
+      Mongoid::QueryCache.enabled = true
+      10.times { Band.create! }
+
+      Band.batch_size(4).all.any?
+    end
+
+    it 'does not cache the result' do
+      expect(Band.all.map(&:id).size).to eq(10)
+    end
```

Comment:
This test is good but there should be another one setting batch size to 4 at query time.

Reply:
I'm not sure I understand what that has to do with the query cache; is the idea to do a query with the same batch size as the number of results of the previous query?

Reply:
Done, as discussed offline